src/Server.jl

```diff
@@ -0,0 +1,431 @@
+                prid = parse(Int, evt.payload["issue"]["number"])
+                pr = pull_request(reponame, prid)
+                sha = pr.head.sha
+                prfiles = pull_request_files(repo, prid)
+                projectfile_found = "Project.toml" in [f.filename for f in prfiles]
+
+                if !projectfile_found
+                    error = "Project file not found on this Pull request"
+                end
+            else
+                ispr = false
+                brn = "master"
+                arg_regx = r"\((.*?)\)"
+                m = match(arg_regx, phrase)
+                if length(m.captures) != 0
+                    arg = strip(m.captures[1])
+                    if length(arg) != 0
+                        brn = arg
+                    end
+                end
+
+                sha, error = get_sha_from_branch(reponame, brn)
+
+                if error == nothing && sha != nothing
+                    gcom = gitcommit(reponame, GitCommit(Dict("sha"=>sha)))
+                    t = tree(reponame, Tree(gcom.tree))
+
+                    for tr in t.tree
+                        if tr["path"] == "Project.toml"
+                            projectfile_found = true
+                            break
+                        end
+                    end
+
+                   if !projectfile_found
+                       error = "Project file not found on branch `$brn`"
+                   end
+                end
+            end
+        else
+            error = "Comment not made by collaborator"
+            report_error = false
+        end
+
+        isvalid = comment_by_collaborator && projectfile_found
+
+        return new(evt, reponame, cloneurl, ispr, prid, sha, brn,
+                  comment_by_collaborator, projectfile_found,
+                  isvalid, error, report_error)
+    end
+end
+
+function get_backtrace(ex)
+    v = IOBuffer()
+    Base.showerror(v, ex, catch_backtrace())
+    return v.data |> copy |> String
+end
+
+function get_jwt_auth()
+    GitHub.JWTAuth(GITHUB_APP_ID, GITHUB_PRIV_PEM)
+end
+
+"""Start a github webhook listener to process events"""
+function start_github_webhook(http_ip=DEFAULT_HTTP_IP, http_port=DEFAULT_HTTP_PORT)
+    auth = get_jwt_auth()
+    listener = GitHub.CommentListener(comment_handler, TRIGGER; check_collab=false, auth=auth, secret=GITHUB_SECRET)
+    GitHub.run(listener, IPv4(http_ip), http_port)
+end
+
+function get_commit(event::WebhookEvent)
+    @info("getting commit sha for event")
+    kind = event.kind
+    payload = event.payload
+    commit = nothing
+    if kind == "push"
+        commit = payload["after"]
+    elseif kind == "pull_request"
+        commit = payload["pull_request"]["head"]["sha"]
+    elseif kind == "status"
+        commit = payload["commit"]["sha"]
+    elseif kind == "pull_request_review"
+        commit = payload["review"]["commit_id"]
+    end
+    commit
+end
+
+event_queue = Queue{RegParams}()
```

Comment:
Are you sure you need a actual `DataStructure.Queue` here?
`pushfirst!` and `pop!` on a `Vector` are more performant than one might expect

Reply:
Yeah, Queue is not really needed here.

Reply:
You will thus be be able to drop the DataStructures.jl dependency

Reply:
This all looks way over my head, but if it still needs to be done, I cold probably help remove DataStructures dependency.

Reply:
https://github.com/JuliaComputing/Registrator.jl/pull/6